src/Raven.Server/ServerWide/SecretProtection.cs

```diff
@@ -319,10 +321,20 @@ private static byte[] DecryptProtectedData(byte[] secret, byte[] key)
             return unprotectedData;
         }
 
-        public RavenServer.CertificateHolder LoadCertificateWithExecutable(string executable, string args, ServerStore serverStore)
+        public enum CertificateRequestType
+        {
+            Load,
+            Renew,
+            CertificateChanged
+        }
+
+        public RavenServer.CertificateHolder LoadCertificateWithExecutable(CertificateRequestType type, string executable, string userArgs, ServerStore serverStore)
         {
             Process process;
 
+            
+            var args = $"{userArgs} {type}";
```

Comment:
Might be better to do the `type` first ? How would that look like to the script?
Because we don't know how many args the user passes, but can say that the first arg to the script is the op type

Reply:
That's a problem. For example if the user does this:
Exec: powershell
Args: -NoProfile /path/to/script

Then if I put the type first, it will run 'powershell type -NoProfile /path/to/script' but type is an argument to the script and not for powershell.

Reply:
Okay, that make me favor having separate config options more for this